playbooks/robusta_playbooks/job_actions.py

```diff
@@ -125,4 +125,108 @@ def job_events_enricher(event: JobEvent, params: EventEnricherParams):
         event.add_enrichment([events_table_block], {SlackAnnotations.ATTACHMENT: True})
 
 
+class JobPodEnricherParams(EventEnricherParams):
+    """
+    :var events: Add the events of the related pod
+    :var logs: Add the logs of the related pod
+    """
+    events: bool = True
+    logs: bool = True
+
+
+@action
+def job_pod_enricher(event: JobEvent, params: JobPodEnricherParams):
+    """
+    Given a Kubernetes job, get information about the latest job pod.
+
+    Used to get the related pod's events and/or logs
+    """
+    job = event.get_job()
+    if not job:
+        logging.error(
+            f"cannot run job_pod_enricher on alert with no job object: {event}"
+        )
+        return
+
+    pod = get_job_latest_pod(job)
+
+    if pod:
+        if params.events:
+            events_table_block = get_resource_events_table(
+                "*Job pod events:*",
+                pod.kind,
+                pod.metadata.name,
+                pod.metadata.namespace,
+                included_types=params.included_types,
+                max_events=params.max_events,
+            )
+            if events_table_block:
+                event.add_enrichment([events_table_block], {SlackAnnotations.ATTACHMENT: True})
+
+        if params.logs:
+            log_data = pod.get_logs()
+            if log_data:
+                event.add_enrichment(
+                    [FileBlock(f"{pod.metadata.name}.log", log_data.encode())],
+                )
+
+
+@action
+def job_info_enricher(event: JobEvent):
+    """
+    Given a Kubernetes job, add information about the job, from the job spec and status.
```

Comment:
Dont know if this is important for this PR or not, just asking to see what you think, maybe add it to notion if you think its out of scope:
is there a way to determine if a rule in `spec.podFailurePolicy.rules` caused the job to fail and which one?
since the job could fail due to bad rules configured like this example where we fail on returning 0
 ```
 podFailurePolicy:
    rules:
    - action: FailJob
      onExitCodes:
        containerName: mycontainer
        operator: In             
        values: [0]

```

Reply:
I wasn't aware of that field.
It's alpha on k8s 1.25
I don't think we need to add it yet